File: apps/python/lib/application/fizz_buzz_list_command.py
from lib.application.fizz_buzz_command import FizzBuzzCommand
from lib.domain.model.fizz_buzz_list import FizzBuzzList
from lib.domain.type.fizz_buzz_type import FizzBuzzType


class FizzBuzzListCommand(FizzBuzzCommand):
    MAX_NUMBER: int = 100

    def __init__(self, type_: FizzBuzzType) -> None:
        self._type = type_

    def execute(self) -> FizzBuzzList:
        values = [self._type.generate(i) for i in range(1, self.MAX_NUMBER + 1)]
        return FizzBuzzList(values)

File: apps/php/src/Application/FizzBuzzListCommand.php
<?php

namespace App\Application;

use App\Domain\Model\FizzBuzzList;
use App\Domain\Type\FizzBuzzType;

final class FizzBuzzListCommand implements FizzBuzzCommand
{
    private const MAX_NUMBER = 100;

    public function __construct(
        private readonly FizzBuzzType $type,
        private readonly int $maxNumber = self::MAX_NUMBER,
    ) {
        if ($maxNumber > self::MAX_NUMBER) {
            throw new \InvalidArgumentException(
                sprintf('最大値は%d以下である必要があります', self::MAX_NUMBER)
            );
        }
    }

    public function execute(int $number = 0): FizzBuzzList
    {
        $maxNumber = $number === 0 ? $this->maxNumber : $number;
        $values = [];

        for ($i = 1; $i <= $maxNumber; $i++) {
            $values[] = $this->type->generate($i);
        }

        return new FizzBuzzList($values);
    }
}
